include/ocpp/common/types.hpp

```diff
@@ -312,6 +312,73 @@ struct Measurement {
     friend std::ostream& operator<<(std::ostream& os, const Measurement& k);
 };
 
+struct DisplayMessageContent {
+    std::string message;
+    std::optional<std::string> language;
+    std::optional<v201::MessageFormatEnum> message_format;
+
+    friend void from_json(const json& j, DisplayMessageContent& m);
+    friend void to_json(json& j, const DisplayMessageContent& m);
+};
+
+struct DisplayMessage {
+    std::optional<int32_t> id;
+    std::optional<v201::MessagePriorityEnum> priority;
+    std::optional<v201::MessageStateEnum> state;
+    std::optional<DateTime> timestamp_from;
+    std::optional<DateTime> timestamp_to;
+    std::optional<std::string> transaction_id;
+    DisplayMessageContent message;
+    std::optional<std::string> qr_code;
+};
+
+struct RunningCostChargingPrice {
+    std::optional<double> kWh_price;
+    std::optional<double> hour_price;
+    std::optional<double> flat_fee;
+
+    friend void from_json(const json& j, RunningCostChargingPrice& c);
+    friend void to_json(json& j, const RunningCostChargingPrice& c);
+};
+
+struct RunningCostIdlePrice {
+    std::optional<uint32_t> idle_grace_minutes;
+    std::optional<double> idle_hour_price;
+
+    friend void from_json(const json& j, RunningCostIdlePrice& c);
+    friend void to_json(json& j, const RunningCostIdlePrice& c);
+};
+
+enum class RunningCostState {
+    Charging,
+    Idle,
+    Finished
+};
+
+namespace conversions {
+RunningCostState string_to_running_cost_state(const std::string& state);
+}
+
+struct RunningCost {
+    std::string transaction_id;
+    std::optional<DateTime> timestamp;
+    std::optional<uint32_t> meter_value;
+    // Transaction cost. This field is not really optional (what is the point of sending RunningCost without cost?),
+    // but it is defined on different places in OCPP 2.0.1 and 1.6. So we can not easily do a conversion from json here.
```

Comment:
Hmm ok I think I can make this field not optional indeed. The problem was that the 'cost' in 1.6 is inside the 'data transfer json' and 2.0.1 it is not inside the 'custom data' json. So in 2.0.1 it must be set separately. But I think this can still work if making it not optional (since there is no auto generated code here :)).